### PR TITLE
Algorithm Peak in 1D Array

### PR DESCRIPTION
**Pull request by PeritusDeveloper**
## Related Issue
- #579

## Proposed Changes
- Added code, algorithm, test cases in **LearnCPP/A-Array** 
- Updated the Readme file with all the changes related to my algorithm.
- Added Examples also.

## Additional Info
- All necessary information is in the Readme file as well as comment present in code file.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

> I have added a code file containing the algorithm for finding the peak in 1D Array.
> I have updated the Readme file with relevant algorithm ,time complexities and examples.
> I have checked my code by taking variety of test cases which also includes the base cases.
> I have successfully submitted the code on [Leetcode] (https://leetcode.com/submissions/detail/668270433/)
> I have tested my cases in Visual Studio Code 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      | Screenshot #2  |
| ----------- | ----------- |
| Test Case 1 | Test Case 2 |
| image | image |

### CODE REVIEW

File: A-Array/README.md [hunk 1]
```diff
@@ -244,3 +245,101 @@ prefix[r]-prefix[l-1]        (O(1) time complexity)
...
+        }
+
+        else if((arr[mid] >= arr[mid-1]) && (arr[mid] >=arr[mid+1])) return mid
+
+        else if(arr[mid] < arr[mid-1]) end=mid-1
+
+        else start=mid+1
+    }
+}
+   ```
```
Comment (Lakhankumawat) [inline A-Array/README.md:336]: remove this code infact you can add psuedocode for problem because code is already included in cpp file

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]
Resolve Merge conflict in your Pr and i will merge it 

your contribution looks good to me

**Comment by Lakhankumawat** [comment]
## Starr the repo click above the red line
Screenshot 2022-03-29 103144

File: A-Array/README.md (hunk 1 above)
Comment (PeritusDeveloper) [reply A-Array/README.md:393]: I will add pseudo code instead

**Comment by PeritusDeveloper** [comment]
I have starred the repo made the necessary changes of writing pseudo code in Readme file.

**Review by Lakhankumawat — APPROVED** [review]
LGTM 🌟🎉

**Comment by Lakhankumawat** [comment]
@all-contributors please add @PeritusDeveloper  code ,

**Pull request merged**